### PR TITLE
Update RichProgressBarTheme after detecting light theme on colab

### LINKED ISSUE #10953: Detect notebook themes & set appropriate progress bar description color while using `RichProgressBar`
## 🐛 Bug

Using the default `RichProgressBar` with the light theme in interactive environments makes it difficult to read progress bar description and metrics.

Screen Shot 2021-12-06 at 3 26 39 PM

### Expected behavior

Switch to contrast colors by detecting the themes.

Could be achieved by fetching the html css selector `theme`.

Screen Shot 2021-12-06 at 3 30 06 PM

### Environment

- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @SeanNaren @kaushikb11

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #10953 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @SeanNaren @kaushikb11

### CODE REVIEW

File: pytorch_lightning/callbacks/progress/rich_progress.py [hunk 1]
```diff
@@ -190,6 +190,23 @@ class RichProgressBarTheme:
     metrics: Union[str, Style] = "white"
 
 
+def _detect_light_colab_theme() -> bool:
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/progress/rich_progress.py:193]: should it use the way we have in tqdm progress bar to check for colab environment?

File: pytorch_lightning/callbacks/progress/rich_progress.py [hunk 2]
```diff
@@ -278,6 +290,19 @@ def test_description(self) -> str:
     def predict_description(self) -> str:
         return "Predicting"
 
+    def _update_for_light_colab_theme(self) -> None:
+        if _detect_light_colab_theme():
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/progress/rich_progress.py:294]: kaggle notebooks has color themes too 😄

File: pytorch_lightning/callbacks/progress/rich_progress.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/callbacks/progress/rich_progress.py:193]: We could. But by using this approach, we could scale this utility for updating themes for jupyter notebooks, kaggle, running on terminal in future, if required.

File: pytorch_lightning/callbacks/progress/rich_progress.py (hunk 2 above)
Comment (kaushikb11) [reply pytorch_lightning/callbacks/progress/rich_progress.py:294]: Screen Shot 2021-12-08 at 5 05 02 PM

Yeah, we could figure it out in a follow up PR by fetching css tags.

**Review by tchaton — APPROVED** [review]
Awesome !

File: pytorch_lightning/callbacks/progress/rich_progress.py (hunk 2 above)
Comment (rohitgr7) [reply pytorch_lightning/callbacks/progress/rich_progress.py:294]: yeah these 3 are the only most popular ones I know: `kaggle`, `colab`, `jupyter-lab (default dark mode)`

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/callbacks/progress/rich_progress.py (hunk 1 above)
Comment (awaelchli) [inline pytorch_lightning/callbacks/progress/rich_progress.py:193]: shouldn't this function be defined at the bottom of the file?

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/callbacks/progress/rich_progress.py [hunk 3]
```diff
@@ -190,6 +190,23 @@ class RichProgressBarTheme:
     metrics: Union[str, Style] = "white"
 
 
+def _detect_light_colab_theme() -> bool:
+    """Detect if it's light theme in Colab."""
```
Comment (carmocca) [inline pytorch_lightning/callbacks/progress/rich_progress.py:194]: Did you find this function somewhere or did you write it yourself? If the former, please add a reference

Comment (kaushikb11) [reply pytorch_lightning/callbacks/progress/rich_progress.py:194]: https://github.com/googlecolab/colabtools/issues/2485

File: pytorch_lightning/callbacks/progress/rich_progress.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/callbacks/progress/rich_progress.py]: Updated!

File: pytorch_lightning/callbacks/progress/rich_progress.py [hunk 4]
```diff
@@ -190,6 +190,23 @@ class RichProgressBarTheme:
...
+    try:
+        get_ipython  # type: ignore
+    except NameError:
+        return False
+    ipython = get_ipython()  # noqa: F821
+    if "google.colab" in str(ipython.__class__):
```
[the remark this answers is not in the document]
Comment (kaushikb11) [reply pytorch_lightning/callbacks/progress/rich_progress.py]: We could add it once we find one more use case for it.

File: tests/callbacks/test_rich_progress_bar.py [hunk 5]
```diff
@@ -284,6 +284,22 @@ def test_rich_progress_bar_counter_with_val_check_interval(tmpdir):
     assert val_bar.completed == 4
     assert val_bar.total == 4
 
+    assert progress_update.call_count == expected_call_count
+
```
Comment (carmocca) [inline tests/callbacks/test_rich_progress_bar.py:288]: Bad merge?

```suggestion
```

**Review by carmocca — APPROVED** [review]

**Pull request merged**